deploy/deploy_test.go

```diff
@@ -68,61 +83,91 @@ func TestKindSpec(t *testing.T) {
 		k: &KindSpec{
 			Name: "test",
 		},
-		execer:  exec.NewFakeExecer(errors.New("cmd failed")),
+		resp: []fexec.Response{
+			{Cmd: "kind", Args: []string{"create", "cluster", "--name", "test"}, Err: "failed to create cluster"},
+		},
 		wantErr: "failed to create cluster",
 	}, {
 		desc: "create cluster with GAR - 1 reg",
 		k: &KindSpec{
 			Name:                     "test",
 			GoogleArtifactRegistries: []string{"us-west1-docker.pkg.dev"},
 		},
-		execer: exec.NewFakeExecer(nil, nil, nil, nil, nil, nil),
+		resp: []fexec.Response{
+			{Cmd: "kind", Args: []string{"create", "cluster", "--name", "test"}},
+			{Cmd: "gcloud", Args: []string{"auth", "print-access-token"}},
+			{Cmd: "docker", Args: []string{"login", "-u", "oauth2accesstoken", "-p", "", "https://us-west1-docker.pkg.dev"}},
+			{Cmd: "kind", Args: []string{"get", "nodes", "--name", "test"}},
+			{Cmd: "docker", Args: []string{"cp", ".*/config.json", ":/var/lib/kubelet/config.json"}},
+			{Cmd: "docker", Args: []string{"exec", "", "systemctl", "restart", "kubelet.service"}},
+		},
 	}, {
 		desc: "create cluster with GAR - 2 regs",
 		k: &KindSpec{
 			Name:                     "test",
 			GoogleArtifactRegistries: []string{"us-west1-docker.pkg.dev", "us-central1-docker.pkg.dev"},
 		},
-		execer: exec.NewFakeExecer(nil, nil, nil, nil, nil, nil, nil),
+		resp: []fexec.Response{
+			{Cmd: "kind", Args: []string{"create", "cluster", "--name", "test"}},
+			{Cmd: "gcloud", Args: []string{"auth", "print-access-token"}},
+			{Cmd: "docker", Args: []string{"login", "-u", "oauth2accesstoken", "-p", "", "https://us-west1-docker.pkg.dev"}},
+			{Cmd: "docker", Args: []string{"login", "-u", "oauth2accesstoken", "-p", "", "https://us-central1-docker.pkg.dev"}},
+			{Cmd: "kind", Args: []string{"get", "nodes", "--name", "test"}},
+			{Cmd: "docker", Args: []string{"cp", ".*/config.json", ":/var/lib/kubelet/config.json"}},
+			{Cmd: "docker", Args: []string{"exec", "", "systemctl", "restart", "kubelet.service"}},
+		},
 	}, {
 		desc: "create cluster with GAR - failed to get access token",
 		k: &KindSpec{
 			Name:                     "test",
 			GoogleArtifactRegistries: []string{"us-west1-docker.pkg.dev"},
 		},
-		execer:  exec.NewFakeExecer(nil, errors.New("failed to get access token")),
+		resp: []fexec.Response{
+			{Cmd: "kind", Args: []string{"create", "cluster", "--name", "test"}, Err: "failed to get access token"},
```

Comment:
this is the cmd that should fail: {Cmd: "gcloud", Args: []string{"auth", "print-access-token"}},

Reply:
Done